calculator.go

```diff
@@ -16,3 +18,12 @@ func Subtract(a, b float64) float64 {
 func Multiply(a, b float64) float64 {
 	return a * b
 }
+
+// Divide takes two numbers and returns the result by dividing the first by the second
+func Divide(a, b float64) (float64, error) {
+	if b != 0 {
+		return a / b, nil
+	}
+
+	return -1, fmt.Errorf("%f is an invalid denominator", b)
```

Comment:
Excellent! In the error case, it obviously doesn't matter what we return as the data value, since the error value signals that it shouldn't be used. The convention, though, is to return zero (don't ask me why, but it's as good as any other choice!)